File: app/src/main/java/com/devx/raju/di/module/WorkModule.java
package com.devx.raju.di.module;


import android.app.Application;

import androidx.annotation.NonNull;
import androidx.work.WorkManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class WorkModule {

    @Provides
    @Singleton
    WorkManager provideWorkManager(@NonNull Application application) {
        return    WorkManager.getInstance(application);
    }

}

File: app/src/main/java/com/devx/raju/di/module/WorkModule.kt
package com.devx.raju.di.module

import android.app.Application
import androidx.work.WorkManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class WorkModule {
    @Provides
    @Singleton
    fun provideWorkManager(application: Application): WorkManager {
        return WorkManager.getInstance(application)
    }
}